### ISSUE TITLE
Mouse right-click cancels selection unless click is inside the selection

### ISSUE THREAD

**Issue by sasumner** [notepad-plus-plus/notepad-plus-plus#8563]
### Description of the Issue

A mouse right-click cancels any currently active selection unless the click is inside the contrasting area defining the selection.  Notepad++ shouldn't cancel the selection if the click is outside.

### Steps to Reproduce the Issue
1. Create some text in a new tab, example:

```
Jack and Jill
Went up the hill
To fetch a pail of water
Jack fell down
And broke his crown,
And Jill came tumbling after.
```

2. Make some sort of selection of that text, example:

![image](https://user-images.githubusercontent.com/30118311/87473617-bc95a200-c5ef-11ea-8793-4e3d8d306980.png)

3. Right-click a spot in the editing window away from the selection, example is where the red arrow points:

![image](https://user-images.githubusercontent.com/30118311/87473594-b6072a80-c5ef-11ea-92b4-c493bcbafb57.png)

### Expected Behavior

I expected my selection to remain intact, regardless of where I right-clicked inside the editor window.  
I expected the context menu to appear, allowing me to copy my selection.
The context menu _does_ appear, but by then my selection is no longer there so that I can copy it.

### Actual Behavior

Right-clicking outside the selection cancels it, moving the caret to the location clicked (if a valid document position).  This can be seen in the screenshot above; the one with the big red arrow.
As left-click moves the caret for me, I don't need right-click to do that.

### Debug Information

Notepad++ v7.8.9   (32-bit)
Build time : Jul  9 2020 - 21:32:00
Path : C:\\.........\npp.7.8.9.bin\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 1809
OS Build : 17763.1039
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll 

### Other

This issue was first raised on the Community site:

https://community.notepad-plus-plus.org/topic/19253/editor-right-click-outside-current-text-selection-lose-this-selection/

**Comment by donho** [comment]
The behaviour you expected was in Scintilla (very) old version.
For me the current behaviour (right-click outside of selection cancels the selection) is more consistence, since it's really "contextual".

**Comment by sasumner** [comment]
Changes from this issue released in Notepad++ version 7.9
